fish-rust/src/future_feature_flags.rs

```diff
@@ -133,38 +106,77 @@ const metadata: [FeatureMetadata; 4] = [
     },
 ];
 
+thread_local!(
+    #[cfg(any(test, feature = "fish-ffi-tests"))]
+    static LOCAL_FEATURES: std::cell::RefCell<Option<Features>> = std::cell::RefCell::new(None);
+);
+
 /// The singleton shared feature set.
-static mut global_features: Features = Features::new();
+static FEATURES: Features = Features::new();
+
+/// Perform a feature test on the global set of features.
+pub fn test(flag: FeatureFlag) -> bool {
+    #[cfg(any(test, feature = "fish-ffi-tests"))]
+    {
+        LOCAL_FEATURES.with(|fc| fc.borrow().as_ref().unwrap_or(&FEATURES).test(flag))
+    }
+    #[cfg(not(any(test, feature = "fish-ffi-tests")))]
+    {
+        FEATURES.test(flag)
+    }
+}
+
+pub use test as feature_test;
+
+/// Set a flag.
+#[cfg(any(test, feature = "fish-ffi-tests"))]
+pub(self) fn set(flag: FeatureFlag, value: bool) {
+    LOCAL_FEATURES.with(|fc| fc.borrow().as_ref().unwrap_or(&FEATURES).set(flag, value));
+}
+
+/// Parses a comma-separated feature-flag string, updating ourselves with the values.
+/// Feature names or group names may be prefixed with "no-" to disable them.
+/// The special group name "all" may be used for those who like to live on the edge.
+/// Unknown features are silently ignored.
+pub fn set_from_string<'a>(str: impl Into<&'a wstr>) {
+    let wstr: &wstr = str.into();
+    #[cfg(any(test, feature = "fish-ffi-tests"))]
+    {
+        LOCAL_FEATURES.with(|fc| {
+            fc.borrow()
+                .as_ref()
+                .unwrap_or(&FEATURES)
+                .set_from_string(wstr)
+        });
+    }
+    #[cfg(not(any(test, feature = "fish-ffi-tests")))]
+    {
+        FEATURES.set_from_string(wstr)
+    }
```

Comment:
BTW I don't think this cfg is built in CI, as if so it should've failed in one of my pushes